downstairs/src/lib.rs

```diff
@@ -1,6 +1,7 @@
 // Copyright 2023 Oxide Computer Company
 #![cfg_attr(usdt_need_asm, feature(asm))]
 #![cfg_attr(all(target_os = "macos", usdt_need_asm_sym), feature(asm_sym))]
+#![warn(clippy::unused_async)]
```

Comment:
Instead of updating all the files directly, should we just put this into the CI script:

```
local:pflush$ git diff .github/workflows/rust.yml 
diff --git a/.github/workflows/rust.yml b/.github/workflows/rust.yml
index 65127d7..e278085 100644
--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -45,4 +45,4 @@ jobs:
     - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4
     - uses: dtolnay/rust-toolchain@1.66
     - name: Test Libraries
-      run: cargo clippy --all-targets
+      run: cargo clippy --all-targets -- -W clippy::unused_async
```

Then, we won't forget to add it to other files and async won't sneak back in?

Reply:
If I used this strategy, it wouldn't fail locally (either running `cargo clippy` or through IDE hints).

I've instead moved this to `.cargo/config`, which enables it for the entire workspace (and should work for CI too).  Once https://github.com/rust-lang/cargo/issues/12115 is released and we bump the toolchain, we can use that instead.

Reply:
Having a local config file was what I actually wanted, I just did not find that solution in my
limited search.

And, if this runs in CI too, then we don't need to update the `.github/workflows/rust.yml ` file
to include it.